### DISCUSSION TITLE
Page and Block level transclusion (logseq embeds, notion sync blocks, etc)

### DISCUSSION THREAD

**Discussion by ThinkSalat** [anyproto/contributors#21] [Anytype apps]
Imagine a card preview
![image](https://github.com/anyproto/contributors/assets/31082405/ddeba8ba-da65-4813-aaa7-4e932bc12644)

Now imagine you can view all the content from that reference block or page by clicking and dragging a handle at the bottom right of the card. You can scroll through the contents of the linked note/block, allowing to view all contents within without displaying the entire document. You can click a button to open the linked page or block as a modal, or you can expand it fully, similarly to how toggles currently work.

I could see simply adding another option under the preview layout menu (or make it easier)
![image](https://github.com/anyproto/contributors/assets/31082405/b4cec64a-fdb8-4f54-b9cb-0b986fa69e3f)

In use, this allows to very quickly edit related material without having to lose context by switching to that note completely - I think note taking and our "on screen" lives suffer from the [Doorway Effect](https://en.wikipedia.org/wiki/Doorway_effect), or at least some version of that. When we have to put attention towards navigation (finding the correct page to put our note), that can easily distract enough to forget or lose the spark of what we had been originally intending to note down. I've been using Logseq a lot recently with their embeds, and it certainly helps to keep focused.

**Comment by JosephASullivan** [comment]
This might be the single most important lacking feature imo.

**Reply by ThinkSalat** [reply]
highly agree. Moved to logseq from anytype when I discovered they have offline first + transclusion. If we got this in anytype I think it'd be a nobrainer to use anytype instead.

**Comment by ThinkSalat** [comment]
I want to bring attention to SiYuan, another offline first open source PKMS that I believe is also just fantastic. It's a pure joy to use and the developers are wonderful people. I want to show you how they've handled transclusion. Regardless of single though sizeable complaint about their system, it's by far the best impelmentation of it and I couldn't have dreamed of a better solution

check it out below. 
![image](https://github.com/anyproto/contributors/assets/31082405/88c5bfcb-39dc-4aea-aaee-829c0bb575b6)

What I like is that the indication that there's transclusion is happening is clear, doesn't take up too much space and doesn't make itself thecenter of attention, as most likely you're going to want to have a more seamless experience. Regardless if you'd like more emphasis on it, you can go ahead and add that with custom css or by simply adding content like a **page title** in bold to the embedded document.

My one gripe is that if you want to edit one of the items that's transcluded, you click on it, it opens a dialog box with that note like a big modal, where you can then do your editing. I would MUCH rather prefer to be able to edit in place and if I so desire, click a button to open the modal, maybe for more intense note taking.

Just thought I'd share just how beautiful and useful it can be!

**Reply by zy55769068** [reply]
Just a reminder: SiYuan has indeed done a good job in this aspect, but I do not recommend using it for the following reasons: The WebDAV service requires mandatory account login, and the author has mentioned that the WebDAV synchronization functionality will incur charges, even for self-hosted WebDAV services. Moreover, the author has a history of embedding mining code in previous projects. Additionally, the migration process involves significant costs since the backend files are in JSON format. During my search for alternatives to Obsidian, I encountered data loss issues with the WebDAV synchronization feature.

Certainly, this part of the functionality is still quite impressive. If Anytype can implement similar features, it would be a great achievement.

**Reply by ThinkSalat** [reply]
Wow, that is not something that I had heard before. That's extremely shady, though SiYuan's source code is available on github, that doesn't mean shady stuff never happens. Their syncing system is the worst of the lot for sure though. I would love to see anytype take their transclusion and run with it. Just allow editing in the trasncluded block instead of forcing open a modal window
